driver/amc_pci_core.c

```diff
@@ -180,7 +254,7 @@ static void destroy_device_nodes(
     int major = amc_priv->major;
     int minor = amc_priv->minor;
 
-    for (int i = 0; i < MINORS_PER_BOARD; i ++)
+    for (int i = 0; i < amc_priv->prom->nentries; i ++)
```

Comment:
Are we sure this won't try to destroy non-existent device nodes if we ignored some entries?

Reply:
The probe function will error if there is an unknown entry, therefore the remove function will not be called in that case